src/client.rs

```diff
@@ -251,11 +251,11 @@ pub struct ApiClient {
 
 impl ApiClient {
     pub fn new(config: Box<ClientConfig>) -> ApiClient {
-        let mut client_builder = reqwest::Client::builder();
+        let client_builder = reqwest::Client::builder();
         // the --insecure CLI flag enables this, to disable the CN name check
-        if config.allows_insecure() {
-            client_builder.danger_disable_hostname_verification();
-        }
+        // if config.allows_insecure() {
+        //     client_builder.danger_accept_invalid_certs(true);
+        // }
```

Comment:
This looks wrong to me, no clue if this was intentional on Abe's part or not...

Reply:
Yeah you should probably undo this... I assume the `danger_accept_invalid_certs` method still exists, so idk why he'd do this aside from forgetting about it during cleanup.

Reply:
They've shuffled these functions around a little bit and the docs give some (rightfully so) strongly worded warnings about using them. What exactly is this insecure flag aiming to accomplish? Why do we have/need a flag to disable the hostname check?

It seems like accepting self-signed certificates would be the main thing we need in this area, not a disabling of hostname/certificate checks, especially without a strongly worded warning of our own.